app/models/user_notification_preferences.rb

```diff
@@ -0,0 +1,64 @@
+class UserNotificationPreferences
+  def initialize(user_id:)
+    @user_id = user_id
+  end
+
+  def id
+    @user_id&.to_i
+  end
+
+  def enabled
+    # course_create and course_update currently always have the same value
+    user_notifications.select(&:course_create).any?
+  end
+
+  def updated_at
+    return nil if user_notifications.empty?
+
+    user_notifications.maximum(:updated_at).iso8601
+  end
+
+  def update(enable_notifications:)
+    UserNotification.transaction do
+      rollback_on_error do
+        UserNotification.where(user_id: id).destroy_all
+
+        user.providers.accredited_body.map do |provider|
```

Comment:
Probably not for this PR but the `providers.accredited_body` is a little confusing. Can this return more than one accrediting body?

Reply:
I see you are destroying and creating rather than updating in the pure sense. Was there a particular reason for this?

Reply:
Yeah.. it is but it's an existing scope. We'll have a look how much it would take to change. There's also another issue with this code as we get duplicates in some cases if the user is related to more than one provider associated with the same accredited body. We'll update 👍

Reply:
Turns out that's a scope from an enum. So an alternative would be `user.providers.where(accrediting_provider: "accredited_body")` which is to my eye even worse 😆 We've left as is for now but added a `distinct` to weed out the duplicates. Hopefully that also makes it a bit more obvious that it's not singular.